### PR TITLE
0.1

### PR DESCRIPTION
**Pull request by meson10**
- Bump up the Redis to not use Redigo, which is not maintained anymore.
- Drop support for Increment, Decrement.
- Unmarshal and Marshal to convert type for In-Memory.
- Expose Keys to list all active keys.

### CODE REVIEW

File: cache_test.go [hunk 1]
```diff
@@ -251,3 +196,44 @@ func testGetMulti(t *testing.T, newCache cacheFactory) {
...
+		t.Errorf("Error in Keys: %s", err)
+	}
+
+	expected := []string{"str", "num", "foo"}
+	if len(items) != len(expected) {
+		t.Errorf("Mismatching number of keys: %v != %v ", items, expected)
+	}
+
+	match := 0
+	for _, i := range expected {
```
Comment (thanhts) [inline cache_test.go:228]: This piece wont work if expected = items = {2; 2}

**Review by sagarrakshe — APPROVED** [review]

File: cache_test.go (hunk 1 above)
Comment (meson10) [reply cache_test.go:228]: Keys won't duplicate themselves in a Hash to begin with.

File: cache.go [hunk 2]
```diff
@@ -95,24 +95,11 @@ type Cache interface {
...
-	//   - an implementation specific error otherwise
-	Decrement(key string, n uint64) (newValue uint64, err error)
-
 	// Expire all cache entries immediately.
 	// This is not implemented for the memcached cache (intentionally).
 	// Returns an implementation specific error if the operation failed.
 	Flush() error
+
+	// Get all currently set keys. This can be super slow so use with care.
+	Keys() ([]string, error)
```
Comment (poorva29) [inline cache.go:104]: Will we ever need this?

Comment (meson10) [reply cache.go:104]: @sagarrakshe and I have required this twice in past 2 days to debug.

**Pull request merged**